helix-view/src/view.rs

```diff
@@ -161,6 +161,35 @@ impl View {
         Some(Position::new(row, col))
     }
 
+    /// Translates a screen position to position in the test document.
+    /// Returns a usize typed position in bounds of the text if found in this view, None if out of view.
+    pub fn pos_at_screen_coords(&self, text: &Rope, row: usize, column: usize) -> Option<usize> {
+        // 2 for status
+        if row < self.area.y as usize || row > self.area.y as usize + self.area.height as usize - 2
+        {
+            return None;
+        }
+
+        let mut pos = text.line_to_char(row - self.area.y as usize + self.first_line);
+
+        // TODO: not ideal
+        const OFFSET: usize = 7; // 1 diagnostic + 5 linenr + 1 gutter
+
+        if column < self.area.x as usize || column > self.area.x as usize + self.area.width as usize
+        {
+            return None;
+        }
+
+        if column > OFFSET + self.area.x as usize {
+            pos += std::cmp::min(
+                column - OFFSET - self.area.x as usize + self.first_col,
+                text.line(text.char_to_line(pos)).len_chars() - 1,
+            );
+        }
+
+        Some(pos)
```

Comment:
This is still not handling graphemes/visual width correctly, you need to fetch the current line, then traverse similar to https://github.com/helix-editor/helix/blob/ba70331b6b9b15d66fc8274b32464643bd9f2a97/helix-view/src/view.rs#L148-L155 until you reach the necessary column. The code in it's current state would treat a tab as four chars for example.

Reply:
Thank you, now I understood at which conditions to test. Will fix